allennlp/nn/util.py

```diff
@@ -54,31 +54,40 @@ def sort_batch_by_length(tensor: torch.autograd.Variable, sequence_lengths: torc
 
     sorted_sequence_lengths, permutation_index = sequence_lengths.sort(0, descending=True)
     sorted_tensor = tensor.index_select(0, permutation_index)
+
+    # This is ugly, but required - we are creating a new variable at runtime, so we
+    # must ensure it has the correct CUDA vs non-CUDA type. We do this by cloning and
+    # refilling one of the inputs to the function.
+    index_range = sequence_lengths.data.clone().copy_(torch.range(0, len(sequence_lengths) - 1))
     # This is the equivalent of zipping with index, sorting by the original
     # sequence lengths and returning the now sorted indices.
-    index_range = Variable(torch.range(0, len(sequence_lengths) - 1).long())
+    index_range = Variable(index_range.long())
     _, reverse_mapping = permutation_index.sort(0, descending=False)
     restoration_indices = index_range.index_select(0, reverse_mapping)
     return sorted_tensor, sorted_sequence_lengths, restoration_indices
 
 
-def get_dropout_mask(dropout_probability: float, shape: List[int]):
+def get_dropout_mask(dropout_probability: float, tensor_for_masking: torch.autograd.Variable):
     """
-    Computes an element-wise dropout mask for an arbitrarily sized tensor, where
+    Computes and returns an element-wise dropout mask for a given tensor, where
     each element in the mask is dropped out with probability dropout_probability.
+    Note that the mask is NOT applied to the tensor.
```

Comment:
You might note that we take the tensor instead of the shape for CUDA vs. non-CUDA issues.